src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -693,13 +705,77 @@ public class DatabaseRecord
 
         public DatabaseTopology Topology;
 
-        public IndexDefinition[] Indexes;
+        public Dictionary<string, IndexDefinition> Indexes;
 
-        public TransformerDefinition[] Transformers;
+        //todo: see how we can protect this
+        public Dictionary<string,TransformerDefinition> Transformers;
 
         public Dictionary<string, string> Settings;
 
         public VersioningConfiguration VersioningConfiguration;
+
+        // todo: see how we can protect this
+        public int LastTransformerId;
+
+        public void AddTransformer(TransformerDefinition definition)
+        {
+            if (Indexes != null && Indexes.Values.Any(x => x.Name == definition.Name))
+            {
+                throw new IndexOrTransformerAlreadyExistException($"Tried to create a transformer with a name of {definition.Name}, but an index under the same name exist");
```

Comment:
Problematic error handling here, we don't want to throw for such errors inside raft